src/main.cpp

```diff
@@ -2,136 +2,114 @@
-    for (byte i = 0; i < 16; i++) {
-        // Compare buffer (= what we've read) with dataBlock (= what we've written)
-        if (buffer[i] == dataBlock[i])
-            count++;
-    }
-    Serial.print(F("Number of bytes that match = ")); Serial.println(count);
-    if (count == 16) {
-        Serial.println(F("Success :-)"));
-        return true;
-    } else {
-        Serial.println(F("Failure, no match :-("));
-        Serial.println(F("  perhaps the write didn't work properly..."));
-        Serial.println();
-        return false;
-    }
-}
 
 void setup() { 
 
   Serial.begin(115200);
 
   // RFID
   SPI.begin(); // Init SPI bus
-  rfid.PCD_Init(); // Init MFRC522 
+  mfrc522.PCD_Init(); // Init MFRC522
+  //mfrc522.PCD_DumpVersionToSerial();  // Show details of PCD - MFRC522 Card Reader details
   for (byte i = 0; i < 6; i++) {
     key.keyByte[i] = 0xFF;
   }
+  Serial.print("Station level is: "); Serial.println(MY_LEVEL);
 
   // Sound
   Serial.println(F("DFRobot DFPlayer Mini Demo"));
   Serial.println(F("Initializing DFPlayer ... (May take 3~5 seconds)"));
   
   mySoftwareSerial.begin(9600);
   if (!myDFPlayer.begin(mySoftwareSerial)) {  //Use softwareSerial to communicate with mp3.
-    Serial.println(F("Unable to begin:"));
+    Serial.println(F("Unable to begin DFPlayer Mini:"));
     Serial.println(F("1.Please recheck the connection!"));
     Serial.println(F("2.Please insert the SD card!"));
-    while(true){
-      delay(0); // Code to compatible with ESP8266 watch dog.
-    }
+    Serial.println(F("Continuing with no sound..."));
+  } else {
+    Serial.println(F("DFPlayer Mini online."));
+    myDFPlayer.volume(VOLUME);  //Set volume value. From 0 to 30
   }
-  Serial.println(F("DFPlayer Mini online."));
-  
-  myDFPlayer.volume(10);  //Set volume value. From 0 to 30
+
+  // LEDs
+  FastLED.addLeds<NEOPIXEL, DATA_PIN>(leds, NUM_LEDS);
+  FastLED.setBrightness(BRIGHTNESS);
 }
  
 void loop() {
 
+  // clearing LEDs if the sound is not playing
+  if (myDFPlayer.readState() != Busy) {
+    FastLED.clear();
+    FastLED.show();
+  };
+
   // Look for new cards
-  if ( ! rfid.PICC_IsNewCardPresent())
+  if ( ! mfrc522.PICC_IsNewCardPresent())
     return;
 
   // Verify if the NUID has been readed
-  if ( ! rfid.PICC_ReadCardSerial())
+  if ( ! mfrc522.PICC_ReadCardSerial())
     return;
 
-  // authenticate!
-  MFRC522::StatusCode auth_status = (MFRC522::StatusCode) rfid.PCD_Authenticate(MFRC522::PICC_CMD_MF_AUTH_KEY_A, trailerBlock, &key, &(rfid.uid));
-  if (auth_status != MFRC522::STATUS_OK) {
-    Serial.print(F("PCD_Authenticate() failed: "));
-    Serial.println(rfid.GetStatusCodeName(auth_status));
+  // perform authentication to open communication
+  auth_success = authenticate(trailerBlock, key);
+  if (!auth_success) {
     return;
   }
 
-  // MIFARE_Read get the size and changes it.
+  // RFID reads get the size and change it.
   // we have to restore it before every call
   size = sizeof(buffer);
```

Comment:
Since we moved the call to a function, which copies the size by value, I think this is no longer relevant